core/src/main/java/com/expedia/adaptivealerting/core/evaluator/RmseEvaluator.java

```diff
@@ -13,16 +11,46 @@
 
     private int n;
     private double resSumSquares;
+    private double rmse;
 
     public RmseEvaluator() {
         this.n = 0;
         this.resSumSquares = 0;
+        this.rmse = 0;
     }
 
     @Override
-    public double computeScore(double observed, double predicted) {
-        this.resSumSquares += MathUtil.getSquare(observed - predicted);
+    public void update(double observed, double predicted) {
+        double residual = observed - predicted;
+        this.resSumSquares += residual * residual;
         this.n += 1;
-        return MathUtil.getSquareRoot(resSumSquares / n);
+        setRmse(Math.sqrt(resSumSquares / n));
+    }
+
+    @Override
+    public double evaluate() {
+        return rmse;
```

Comment:
Perform the calculation here, as opposed to doing it in the update method and returning the stored value. That way you do the calculation only when it's actually needed.